src/shared/user-record-show.c

```diff
@@ -28,14 +28,19 @@ const char* user_record_state_color(const char *state) {
         return NULL;
 }
 
-static void dump_self_modifiable(const char *heading, char **field, const char **value) {
+static void dump_self_modifiable(
+                UserRecord *hr,
+                const char *heading,
+                char **field,
+                const char **value) {
+
         assert(heading);
 
         /* Helper function for printing the various self_modifiable_* fields from the user record */
 
         if (strv_isempty((char**) value))
                 /* Case 1: the array is explicitly set to be empty by the administrator */
-                printf("%13s %sDisabled by Administrator%s\n", heading, ansi_highlight_red(), ansi_normal());
+                printf("%13s %sdisabled by Administrator%s\n", heading, user_record_disposition(hr) == USER_REGULAR ? ansi_highlight_red() : ansi_normal(), ansi_normal());
```

Comment:
Hmm, well in this case the setting wasn't disabled by the admin.

Really there's probably 4 cases:

- Default regular: field is null but value isn't empty
- Default other: field is null and value is empty
- Disabled by admin: field is nonnull but not empty
- Custom list: field is nonempty